# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaInfo.h
# Commit: 279a5522919a
# Full Hash: 279a5522919ac625c1f35151dc8d70fccfb4b8a1
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Add the ability to print a VideoInfo and a VideoDecoderConfigInternal. r=chunmin,media-playback-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D196336
# ==============================================================================

diff -r d43f0c82f214 -r 279a5522919a dom/media/MediaInfo.h
--- a/dom/media/MediaInfo.h	Fri Dec 22 21:39:16 2023 +0000
+++ b/dom/media/MediaInfo.h	Fri Dec 22 21:39:16 2023 +0000
@@ -424,6 +424,81 @@
     }
   }
 
+  nsString ToString() const {
+    std::array YUVColorSpaceStrings = {"BT601", "BT709", "BT2020", "Identity",
+                                       "Default"};
+
+    std::array ColorDepthStrings = {
+        "COLOR_8",
+        "COLOR_10",
+        "COLOR_12",
+        "COLOR_16",
+    };
+
+    std::array TransferFunctionStrings = {
+        "BT709",
+        "SRGB",
+        "PQ",
+        "HLG",
+    };
+
+    std::array ColorRangeStrings = {
+        "LIMITED",
+        "FULL",
+    };
+
+    std::array ColorPrimariesStrings = {"Display",
+                                        "UNKNOWN"
+                                        "SRGB",
+                                        "DISPLAY_P3",
+                                        "BT601_525",
+                                        "BT709",
+                                        "BT601_625"
+                                        "BT709",
+                                        "BT2020"};
+    nsString rv;
+    rv.AppendLiteral(u"VideoInfo: ");
+    rv.AppendPrintf("display size: %dx%d ", mDisplay.Width(),
+                    mDisplay.Height());
+    rv.AppendPrintf("stereo mode: %d", static_cast<int>(mStereoMode));
+    rv.AppendPrintf("image size: %dx%d ", mImage.Width(), mImage.Height());
+    if (mCodecSpecificConfig) {
+      rv.AppendPrintf("codec specific config: %zu bytes",
+                      mCodecSpecificConfig->Length());
+    }
+    if (mExtraData) {
+      rv.AppendPrintf("extra data: %zu bytes", mExtraData->Length());
+    }
+    rv.AppendPrintf("rotation: %d", static_cast<int>(mRotation));
+    rv.AppendPrintf("colors: %s", ColorDepthStrings[static_cast<int>(mColorDepth)]);
+    if (mColorSpace) {
+      rv.AppendPrintf(
+          "YUV colorspace: %s ",
+          YUVColorSpaceStrings[static_cast<int>(mColorSpace.value())]);
+    }
+    if (mColorPrimaries) {
+      rv.AppendPrintf(
+          "color primaries: %s ",
+          ColorPrimariesStrings[static_cast<int>(mColorPrimaries.value())]);
+    }
+    if (mTransferFunction) {
+      rv.AppendPrintf(
+          "transfer function %s ",
+          TransferFunctionStrings[static_cast<int>(mTransferFunction.value())]);
+    }
+    rv.AppendPrintf("color range: %s", ColorRangeStrings[static_cast<int>(mColorRange)]);
+    if (mImageRect) {
+      rv.AppendPrintf("image rect: %dx%d", mImageRect->Width(),
+                      mImageRect->Height());
+    }
+    rv.AppendPrintf("alpha present: %s", mAlphaPresent ? "true" : "false");
+    if (mFrameRate) {
+      rv.AppendPrintf("frame rate: %dHz", mFrameRate.value());
+    }
+
+    return rv;
+  }
+
   // Size in pixels at which the video is rendered. This is after it has
   // been scaled by its aspect ratio.
   gfx::IntSize mDisplay;